h/security/policy/_api_cookie.py

```diff
@@ -0,0 +1,34 @@
+from pyramid.request import Request
+from pyramid.security import Allowed, Denied
+
+from h.security.identity import Identity
+from h.security.policy._cookie import CookiePolicy
+
+COOKIE_AUTHENTICATABLE_API_REQUESTS = [
+    ("api.groups", "POST"),  # Create a new group.
+    ("api.group", "PATCH"),  # Edit an existing group.
+]
+
+
+class APICookiePolicy:
+    """Authenticate API requests with cookies."""
+
+    def __init__(self, cookie_policy: CookiePolicy):
+        self.cookie_policy = cookie_policy
+
+    @staticmethod
+    def handles(request: Request) -> bool:
+        """Return True if this policy applies to `request`."""
+        return (
+            request.matched_route.name,
+            request.method,
+        ) in COOKIE_AUTHENTICATABLE_API_REQUESTS
```

Comment:
The h session cookie currently uses `SameSite=Lax`. This means that the cookies will be set on requests coming from h pages, but also on navigations from other websites to h if the request uses a "safe" (non-mutating) HTTP method (per https://datatracker.ietf.org/doc/html/draft-west-first-party-cookies-07#section-4.1.1). This means that a website could create a link or form which triggers a `GET` request that allows cookie auth, but not a POST/PATCH requests like the groups requests added here.

If we wanted to be more secure, we could use a separate `SameSite=strict` cookie for this auth, or add a check on the `Origin` header.

Reply:
> It sounds like it's doable but perhaps it can be a follow-up PR? 

Yes, I think that makes sense.